core/auth/src/main/java/software/amazon/awssdk/auth/credentials/internal/StaticResourcesEndpointProvider.java

```diff
@@ -58,6 +59,11 @@ public Map<String, String> headers() {
         return Collections.unmodifiableMap(headers);
     }
 
+    @Override
+    public ResourcesEndpointRetryPolicy retryPolicy() {
+        return new ContainerCredentialsRetryPolicy();
+    }
```

Comment:
I don't think we can add retry at this point for IMDS credentials provider because it may increase latency and break customers.

Reply:
right, that was a ask from Hadoop. Instead we can add it as an optional config as well